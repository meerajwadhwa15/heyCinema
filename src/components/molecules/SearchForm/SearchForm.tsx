import React, { FunctionComponent, SyntheticEvent } from 'react';
import { DebounceInput } from 'react-debounce-input';

export type Props = {
  searchResults: (val: string) => void;
};

const SearchForm: FunctionComponent<Props> = ({ searchResults }) => (
  <div className="search-form">
    <form>
      <DebounceInput
        placeholder="Search"
        minLength={2}
        debounceTimeout={300}
        onChange={event => searchResults(event.target.value)}
      />
    </form>
  </div>
);

export default SearchForm;
