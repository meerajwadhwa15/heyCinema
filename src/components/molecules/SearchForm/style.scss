.search-form {
  background: $secondaryBackgroundColor;
  padding: 10px;
  margin-bottom: 20px;

  input {
    padding: 8px 5px;
  }
}
