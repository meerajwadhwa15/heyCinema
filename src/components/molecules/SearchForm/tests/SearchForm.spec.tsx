import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import SearchForm, { Props } from '../SearchForm';

const props: Props = {
  searchResults: jest.fn()
};

describe('SearchForm Enzyme testing', () => {
  test('renders without crashing', () => {
    shallow(<SearchForm {...props} />);
  });
});

describe('SearchForm Snapshot testing', () => {
  test('Matches the snapshot', () => {
    const component = create(<SearchForm {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
