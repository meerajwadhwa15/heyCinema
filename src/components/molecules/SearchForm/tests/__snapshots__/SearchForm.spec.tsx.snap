// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm Snapshot testing Matches the snapshot 1`] = `
<div
  className="search-form"
>
  <form>
    <input
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
  </form>
</div>
`;
